File: e-wallet-fe/src/component/wallet/transaction/Transaction.js
import CollapsibleAction from 'component/wallet/action/CollapsibleAction';

const Transaction = ({transaction}) => {

  return (
    <CollapsibleAction key={transaction.id} label={`${transaction.transactionDate} - ${transaction.description} - ${transaction.amount}`}>
      <div>
        {transaction.amount ? <div>Amount: {transaction.amount}</div> : null}
        {transaction.balance ? <div>Balance: {transaction.balance}</div> : null}
        {transaction.transactionDate ? <div>Transaction date: {transaction.transactionDate}</div> : null}
      </div>
    </CollapsibleAction>
  );
};

export default Transaction;

File: e-wallet-be/src/main/java/com/wallet/dao/entity/Transaction.java
package com.wallet.dao.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Transaction")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid2")
    private String id;

    @Size(max = 250)
    private String description;

    private BigDecimal amount;

    private BigDecimal balance;

    private LocalDateTime transactionDate;

    String walletId;

    @Size(max=50)
    String username;

}
